### DISCUSSION TITLE
is there any way to make trio optional while using AnyIO with pytest

### DISCUSSION THREAD

**Discussion by nikhil-gofynd** [agronholm/anyio#407] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/56588673/147950549-161cf4fb-84d1-4a64-8cb8-4c9b873c519d.png)

can anyone help me with this, I do not want to use trio in the project, Want to use AnyIO for async test cases.
Got the reference from here https://fastapi.tiangolo.com/advanced/async-tests/

**Comment by agronholm** [comment, marked as answer]
Have you tried looking it up in the [documentation](https://anyio.readthedocs.io/en/stable/testing.html#specifying-the-backends-to-run-on)?

**Reply by nikhil-gofynd** [reply]
This seems to be an interesting implementation, where if the backend is not specified use all the backends, rather than sticking with any one of them.

Anyways Thanks

**Reply by agronholm** [reply]
AnyIO libraries would need to test against all backends, while applications should pick one and stick with it. The default was made this way so that test suites would not erroneously pass if one of the supported backend was not present.

**Reply by nikhil-gofynd** [reply]
Got it, Thanks for the quick response.

**Comment by sanya1998** [comment]
смог решить проблему без установки trio: `poetry add "anyio[asyncio]"`
